### PR TITLE
fix null reference exception when re-enabling input system

### PR DESCRIPTION
**Pull request by davidkline-ms**
When the input system initializes, it instantiates the registered gaze provider. When it is disabled, the gaze provider is destroyed. Unfortunately, enable was not re-creating the gaze provider, resulting in null reference exceptions.

This change creates the InstantiateGazeProvider method and calls it from Initialize and, if null, from Enable.

Fixes: #5085

## Verification
Create simple scene in editor. 
Apply the script shown below to any object.
Disable and then re-enable the input system.
Use the input simulation provider to display a hand (ex: spacebar)

```c#
using Microsoft.MixedReality.Toolkit;
using Microsoft.MixedReality.Toolkit.Input;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class EnableDisableSystemTest : MonoBehaviour
{
    private IMixedRealityInputSystem inputSystem = null;
    private IMixedRealityInputSystem InputSystem
    {
        get
        {
            if (inputSystem == null)
            {
                MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem);
            }
            return inputSystem;
        }
    }

    public void DisableInputSystem()
    {
        InputSystem.Disable();
    }

    public void EnableInputSystem()
    {
        InputSystem.Enable();
    }

    public void Update()
    {
        if (Input.GetKeyDown(KeyCode.Alpha1))
        {
            Debug.Log("Enabling input system");
            EnableInputSystem();
        }
        if (Input.GetKeyDown(KeyCode.Alpha2))
        {
            Debug.Log("Disabling input system");
            DisableInputSystem();
        }
    }
}
```

### CODE REVIEW

File: Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs [hunk 1]
```diff
@@ -228,11 +217,34 @@ public override void Initialize()
...
         }
 
+        private void InstantiateGazeProvider(MixedRealityPointerProfile pointerProfile)
+        {
+            if (pointerProfile?.GazeProviderType?.Type != null)
+            {
+                GazeProvider = CameraCache.Main.gameObject.EnsureComponent(pointerProfile.GazeProviderType.Type) as IMixedRealityGazeProvider;
+                GazeProvider.GazeCursorPrefab = pointerProfile.GazeCursorPrefab;
+                // Current implementation implements both provider types in one concrete class.
+                EyeGazeProvider = GazeProvider as IMixedRealityEyeGazeProvider;
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs]: Worth adding a check if EyeGazeProvider null and output warning? 

This seems like a hidden thing that the GazeProvider needs to implement both interfaces

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs]: EyeGazeProvider and GazeProvider are the same thing, but i do agree.

**Comment by davidkline-ms** [comment]
> This seems like something that would greatly benefit from a test, could you please include that in your PR?

Test added. Confirmed fails when the exception is thrown, and passes with the fix.

**Pull request merged**